### ISSUE TITLE
Ability to add files to the template without code changes

### ISSUE THREAD

**Issue by deeptowncitizen** [OpenAPITools/openapi-generator#5219]
I'm using `typescript-fetch` template to generate TS api client.
In addition to all standard files I want to generate actions and reducers for my redux code.
Correct me if I'm wrong, but the current template does not support any additional files, so I can't simply add a new file e.g. `apis.actions.mustache` and use it to generate the file in apis folder?

Is that possible to add some sort of references (e.g. partials from mustache) to automatically resolve templates from files?

**Comment by amakhrov** [comment]
I don't believe openapi-generator allows to compile arbitrary external mustache templates at the moment.

However, I would imagine that it can be universally supported across generators by creating a cli option for each bag of templates (api templates, model templates, supporting files). E.g. `--api-template apis.actions.mustache --api-template another.mustache --model-template my-model-extension.mustache". Or maybe a json config params instead of cli options.

It arises some more questions, though.
- What should the semantics of that be: "add new template(s)" or "replace built-in templates with the provided"? 
- Does it only make sense when paired with `--template-dir` option, with a requirement for the new templates to reside in that dir?
- What should be the filename emitted for each kind of templates? Apparently it cannot be the same as regular ones, otherwise original generated files would be overwritten. Yet another configuration for naming schema then?

However, with a concise interface addressing the above, I definitely see the value of such extra flexibility.
